### PR TITLE
Continue adding tests

### LINKED ISSUE #22: Null `StateStoreGetResponse.ContentType` causes `ArgumentNullException` in simulated bulk get operations
## Expected Behavior

In a state store component that does *not* implement `IBulkStateStore`, the `StateStoreAdaptor` simulates bulk get operations by calling the store's "get" method sequentially.  Responses are then transformed into the expected `BulkGetResponse` type.  If the `StateStoreGetResponse.ContentType` property is omitted, it should result in the bulk response having an empty (i.e. "") `ContentType` property.

## Actual Behavior

An `ArgumentNullException` is thrown during the transformation process.

## Steps to Reproduce the Problem

1. Implement and register a state store (i.e. `IStateStore`) but *do not* implement `IBulkStateStore`.
2. Have the state store's `GetAsync()` method return a `StateStoreGetResponse` that omits the `ContentType` property.
3. Have the component's client call the `BulkGet` method.
4. Note the exception.

## Workaround

Always set the `ContentType` property or implement `IBulkStateStore` (to avoid simulation of bulk operations).

## Release Note

RELEASE NOTE:

### PR DESCRIPTION
**Pull request by philliphoff**
# Description

More tests, primarily focused on the bulk, transactionable, and queryable state store related types.  Found and fixed #22. 

## Issue reference

We strive to have all PR being opened based on an issue, where the problem or feature have been discussed prior to implementation.

Please reference the issue this PR will close: #22 

## Checklist

Please make sure you've completed the relevant tasks for this PR, out of the following list:

* [x] Code compiles correctly
* [x] Created/updated tests
* [ ] Extended the documentation

### CODE REVIEW

File: src/Dapr.PluggableComponents.Tests/Components/StateStore/StateStoreTransactOperationTests.cs [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+        Assert.Equal(key, request.Key);
+    }
+
+    [Fact]
+    public void FromTransactionalStateOperationUnknown()
+    {
+        var grpcOperation = new TransactionalStateOperation();
+
+        Assert.Throws<ArgumentOutOfRangeException>(() => StateStoreTransactOperation.FromTransactionalStateOperation(grpcOperation));
+    }
```
Comment (tmacam) [inline src/Dapr.PluggableComponents.Tests/Components/StateStore/StateStoreTransactOperationTests.cs:67]: Nice

File: src/Dapr.PluggableComponents.Tests/Adaptors/TestServerCallContext.cs [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ------------------------------------------------------------------------
+
+using Grpc.Core;
+
+namespace Dapr.PluggableComponents.Adaptors;
+
+internal sealed class TestServerCallContext : ServerCallContext, IDisposable
+{
```
Comment (tmacam) [inline src/Dapr.PluggableComponents.Tests/Adaptors/TestServerCallContext.cs:19]: Isn't there a class like this defined by gRPC .NET sdk?

File: src/Dapr.PluggableComponents.Tests/Components/StateStore/StateStoreGetResponseTests.cs [hunk 3]
```diff
@@ -18,107 +18,89 @@ namespace Dapr.PluggableComponents.Components.StateStore;
...
+
+        ConversionAssert.ETagEqual(
+            etag => new StateStoreGetResponse { ETag = etag },
+            converter,
+            response => response.Etag);
+
+        ConversionAssert.MetadataEqual(
+            metadata => new StateStoreGetResponse { Metadata = metadata },
+            converter,
+            response => response.Metadata);
```
Comment (tmacam) [inline src/Dapr.PluggableComponents.Tests/Components/StateStore/StateStoreGetResponseTests.cs:54]: nice

File: src/Dapr.PluggableComponents.Tests/Components/StateStore/StateStoreSetRequestTests.cs [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+public sealed class StateStoreSetRequestTests
+{
+    [Fact]
+    public void FromSetRequestConversionTests()
+    {
+        var converter = StateStoreSetRequest.FromSetRequest;
+        StateStoreSetRequest curriedConverter<T>(T _, SetRequest request) => converter(request);
+
+        // NOTE: Conversion from gRPC request content type is not symmetric with the reverse,
+        //       so one cannot use the ConversionAssert.ContentTypeEqual() method.
```
Comment (tmacam) [inline src/Dapr.PluggableComponents.Tests/Components/StateStore/StateStoreSetRequestTests.cs:29]: 👍

**Review by tmacam — APPROVED** [review]

**Comment by tmacam** [comment]
LGTM

File: src/Dapr.PluggableComponents.Tests/Adaptors/TestServerCallContext.cs (hunk 2 above)
Comment (philliphoff) [reply src/Dapr.PluggableComponents.Tests/Adaptors/TestServerCallContext.cs:19]: `ServerCallContext` is `abstract` and the gRPC SDK has its own implementation, but unfortunately it's not exposed to clients.

File: src/Dapr.PluggableComponents.Tests/Components/StateStore/StateStoreGetResponseTests.cs (hunk 3 above)
Comment (philliphoff) [reply src/Dapr.PluggableComponents.Tests/Components/StateStore/StateStoreGetResponseTests.cs:54]: I originally thought it'd end up being much more tedious, but it works reasonably well; I'll likely (eventually) refactor some of the previous rounds of tests to use this model.  The main issues is that conversion isn't always symmetric (e.g. .NET SDK uses `string?` but the generated `.proto` uses `string`) which requires some duplication.

**Pull request merged**